internal/commands/push.go

```diff
@@ -69,10 +71,20 @@ func runPush(dockerCli command.Cli, name string) error {
 		return err
 	}
 
-	cnabRef := reference.TagNameOnly(ref)
-
 	// Push the bundle
-	return pushBundle(dockerCli, bndl, cnabRef)
+	dg, err := pushBundle(dockerCli, bndl, ref)
+	if err != nil {
+		return errors.Wrapf(err, "could not push %q", ref)
+	}
+
+	// we can't just re-use bndl var here, as fixup did rewrite the bundle
```

Comment:
As stated in the specs, https://github.com/cnabio/cnab-spec/blob/master/101-bundle-json.md#invocation-images 
> During bundle development, it may be ideal to omit the contentDigest field and/or skip validation. Once a bundle is ready to be transmitted as a thick or thin bundle, it must have a contentDigest field. 

the contentDigest can be omitted, but needs to be filled before pushing. I think that's exactly what we do here, that's why the bundle is mutated by the `push`.